plutus-ledger/src/Ledger/Fee.hs

```diff
@@ -74,6 +109,245 @@ makeAutoBalancedTransaction params utxo (CardanoBuildTx txBodyContent) pChangeAd
       Nothing
 
 
+-- | Creates a balanced transaction by calculating the execution units, the fees and then the balance.
+-- If the balance is negative the utxo provider is asked to pick extra inputs to make the balance is positive,
+-- which is then assigned to the change address.
+-- The collateral is similarly balanced.
+-- Unlike `makeAutoBalancedTransaction` this function also balances non-Ada.
+makeAutoBalancedTransactionWithUtxoProvider
```

Comment:
Should we delete `makeAutoBalancedTransaction` if this one is more complete and balances non-Ada?

Reply:
I can imagine that you would not want automatic non-Ada balancing. It could hide bugs.